src/ctelnet.cpp

```diff
@@ -356,28 +356,28 @@ void cTelnet::handle_socket_signal_disconnected()
 void cTelnet::handle_socket_signal_hostFound(QHostInfo hostInfo)
 {
     if (!hostInfo.addresses().isEmpty()) {
-        mHostAddress = hostInfo.addresses().first();
+        mHostAddress = hostInfo.addresses().constFirst();
         QString msg = "[ INFO ]  - The IP address of " + hostName + " has been found. It is: " + mHostAddress.toString() + "\n";
         postMessage(msg);
         msg = "[ INFO ]  - Trying to connect to " + mHostAddress.toString() + ":" + QString::number(hostPort) + " ...\n";
         postMessage(msg);
         socket.connectToHost(mHostAddress, hostPort);
     } else {
         socket.connectToHost(hostInfo.hostName(), hostPort);
-        QString msg = "[ ERROR ] - Host name lookup Failure!\nConnection cannot be established.\nThe server name is not correct, not working properly,\nor your nameservers are not working properly.";
+        QString msg = QStringLiteral("[ ERROR ] - Host name lookup Failure!\nConnection cannot be established.\nThe server name is not correct, not working properly,\nor your nameservers are not working properly.");
         postMessage(msg);
         return;
     }
 }
 
 bool cTelnet::sendData(QString& data)
 {
-    while (data.indexOf("\n") != -1) {
-        data.replace(QChar('\n'), "");
+    while (data.indexOf(QStringLiteral("\n")) != -1) {
+        data.replace(QChar('\n'), QString());
```

Comment:
This should be `data.remove(QChar::LineFeed);` IMHO.

Reply:
Good call

Reply:
The while loop is entirely unnecessary then, so I've removed it - `.remove()` is recursive. Could you confirm?